src/main/java/de/zalando/sprocwrapper/SProcCall.java

```diff
@@ -41,10 +41,35 @@ public long getSprocId() {
     }
 
     public static enum WriteTransaction {
-        USE_FROM_SERVICE,
-        NONE,
-        ONE_PHASE,
-        TWO_PHASE
+        USE_FROM_SERVICE {
+            @Override
+            public SProcService.WriteTransaction getServiceWriteTransaction(SProcService.WriteTransaction serviceWriteTransaction) {
+                if (serviceWriteTransaction == null) {
```

Comment:
Another question is if it was a good idea from the beginning to have two `WriteTransaction` enums with slightly different contents...